app/models/concerns/scholars_archive/date_operations.rb

```diff
@@ -0,0 +1,134 @@
+module ScholarsArchive
+  module DateOperations
+    extend ActiveSupport::Concern
+
+    def to_solr(solr_doc={})
+      solr_doc = super
+      solr_doc = solr_doc.merge({"date_decades_ssim" => decades})
+      solr_doc = solr_doc.merge({"date_facet_yearly_ssim" => date_facet_yearly})
+      solr_doc
+    end
+
+    def date_facet_yearly
+      return nil if yearly_dates.empty?
+      yearly_dates
+    rescue ArgumentError
+    end
+
+    def decades
+      return nil if decade_dates.empty?
+      decade_dates.map(&:decade)
+    rescue ArgumentError
+    end
+
+    # Determine the date value to use for Decades facet and date facet yearly processing.
+    def date_value
+      if date_created.present? then
+        date_created
+      elsif date_copyright.present? then
+        date_copyright
+      elsif date_issued.present? then
+        date_issued
+      end
+    end
+
+    def yearly_dates
+      (date_value) ? clean_years : []
+    end
+
+    def clean_years
+      date = Date.edtf(date_value)
+      if date.instance_of? EDTF::Interval
+        date.map(&:year).uniq
+      elsif date.instance_of? Date
+        Array.wrap(date.year)
+      else
+        Array.wrap(clean_datetime.year)
```

Comment:
Because in both places you're using `clean_datetime.year`, it seems to be more clear if the method would return the year. For instance, 

`def parsed_year`